moonraker/components/sensor.py

```diff
@@ -0,0 +1,309 @@
+                if sensor_class is None:
+                    raise config.error(f"Unsupported sensor type: {sensor_type}")
+
+                self.sensors[name] = sensor_class(
+                    name=name,
+                    cfg=cfg,
+                    store_size=self.store_size,
+                )
+            except Exception as e:
+                # Ensures that configuration errors are shown to the user
+                self.server.add_warning(
+                    f"Failed to configure sensor [{cfg.get_name()}]\n{e}"
+                )
+                continue
+
+    def _update_sensor_values(self, eventtime: float) -> float:
+        """
+        Iterate through the sensors and store the last updated value.
+        """
+        changed_data: Dict[str, Dict[str, Union[int, float]]] = {}
+        for sensor_name, sensor in self.sensors.items():
+            base_value = sensor.last_value
+            sensor._update_sensor_value(eventtime=eventtime)
+
+            # Notify if a change in sensor values was detected
+            if base_value != sensor.last_value:
+                changed_data[sensor_name] = sensor.last_value
+        if changed_data:
+            self.server.send_event(SENSOR_EVENT_NAME, changed_data)
+
+        return eventtime + SENSOR_UPDATE_TIME
+
+    async def component_init(self) -> None:
+        try:
+            logging.debug("Initializing sensor component")
+            for sensor in self.sensors.values():
+                if not await sensor.initialize():
+                    self.server.add_warning(
+                        f"Sensor '{sensor.get_name()}' failed to initialize"
+                    )
+
+            self.sensors_update_timer.start()
+
+        except Exception as e:
+            logging.exception(e)
+
+    async def _handle_sensor_list_request(
+        self, web_request: WebRequest
+    ) -> Dict[str, Dict[str, Any]]:
+        output = {
+            "sensors": {
+                key: sensor.get_sensor_info() for key, sensor in self.sensors.items()
+            }
+        }
+        return output
+
+    async def _handle_sensor_info_request(
+        self, web_request: WebRequest
+    ) -> Dict[str, Any]:
+        sensor_name: str = web_request.get_str("sensor")
+        if sensor_name not in self.sensors:
+            raise self.server.error(f"No valid sensor named {sensor_name}")
+
+        sensor = self.sensors[sensor_name]
+
+        return sensor.get_sensor_info()
+
+    async def _handle_sensor_measurements_request(
+        self, web_request: WebRequest
+    ) -> Dict[str, Dict[str, Any]]:
+        sensor_name: str = web_request.get_str("sensor", "")
+        if sensor_name and sensor_name not in self.sensors:
+            raise self.server.error(f"No valid sensor named {sensor_name}")
+
+        output = {
+            key: sensor.get_sensor_measurements()
+            for key, sensor in self.sensors.items()
+            if sensor_name is None or key == sensor_name
```

Comment:
I may have missed this on the previous review, however `sensor_name` defaults to a null string on line 290.  This will result in an empty return value when a front end requests all measurements.

Reply:
Sorry, missed updating the condition when mypy complained about the None as sentinel value when retrieving the request param.